nomenclature/code.py

```diff
@@ -194,3 +194,18 @@ def agg_kwargs(self) -> Dict[str, Any]:
             if self.region_aggregation is not None
             else self.pyam_agg_kwargs
         )
+
+
+class RegionCode(Code):
+    """A subclass of Code specified for regions
+
+    Attributes
+    ----------
+    name : str
+        Name of the RegionCode
+    hierarchy : str
+        Hierarchy of the RegionCode
+
+    """
+
+    hierarchy: Optional[str] = None
```

Comment:
This should be a required attribute.
```suggestion
    hierarchy: str
```